### DISCUSSION TITLE
Cutting diagram mapping

### DISCUSSION THREAD

**Discussion by danbaehr** [lairdubois/lairdubois-opencutlist-sketchup-extension#389] [Q&A] [answered]
It's difficult to match cutting diagram part IDs with the associated parts on the model in Sketchup. It appears that the only way is to click, one by one, on the magnifying glass icon for each part so that it gets highlighted on the model. Is there any way to produce a mapping diagram that shows the model with part IDs overlayed? I couldn't think of how that could be done via the plugin but II thought I'd ask.

My use case:
I'm designing a deck. I was able to use OpenCutlist to generate an optimized deck board list and cutting diagram, which worked great. However, the part IDs don't seem to follow any kind of sorting with the model and the cutting diagram is only sorted according to efficiency. In other words the part IDs seem to be randomly assigned to different boards in my deck design, rather than flowing A-Z from left to right. Given that the cutting diagram is also not organized from A-Z, it makes it difficult to match an ID with a board in the model.

**Comment by bbeaulant** [comment, marked as answer]
Hi @danbaehr,

Currently there's not automatic way to generate some labels on the 3D model that point to each part with its ID.

Parts in cutting diagrams are by definition sorted to be efficient.
And if part number is not enough to know where the part is, you can display their name instead.

<img width="1212" alt="Capture d’écran 2022-03-08 à 07 44 54" src="https://user-images.githubusercontent.com/156056/157182069-5c633fa8-38b2-4d5b-8a43-e981285d6775.png">

<img width="1212" alt="Capture d’écran 2022-03-08 à 07 45 12" src="https://user-images.githubusercontent.com/156056/157182058-6bf76c81-dd97-46ed-a32b-eb5bef8ed114.png">

Regards

**Comment by danbaehr** [comment]
Hi Boris! Thanks for the quick reply!

I see what you're saying. I could go through and label each deck board, in order, and then use the designation to view them. That would help, for sure. I have 85 boards so it would be a bit time consuming to label each one but that might be my best approach. If there's ever a way to label the model components automatically that would be preferred but I can submit that as an RFE for future consideration. 

Thanks for the help!
